### PR TITLE
Feature/hierarchy builder

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Integrate hierarchy builder into the import process:

* Produce `data-import-complete` messages for each hierarchy when all observations are imported.
* Consume `hierarchy-built` messages and update the task state on the instance (on the dataset-api)
* Determine when all hierarchies are built and set the instance to complete
* Update go version to 1.9.2 in build configs
* Added Kafka consumer group environment variables to be consistent with other repos
* Left todo's where I think we need to integrate the error reporter into the tracker
* Update config to use json based logging with sensitive fields omitted.
* Moved common events / schemas into dp-import repo, and vendored them into this repo.

I think it would be good to start breaking the tracker out from main into separate packages to facilitate unit testing.

### How to review

Review code

Can be integration tested with these repo's on the same branch name:
dp-recipe-api
dp-import-api
dp-dataset-api
dp-hierarchy-builder
dp-import-tracker

### Who can review

Anyone

### CODE REVIEW

File: cmd/dp-import-tracker/main.go [hunk 1]
```diff
@@ -168,58 +212,92 @@ func manageActiveInstanceEvents(
 		case <-tickerChan:
 			log.Debug("check import Instances", log.Data{"q": trackedInstances})
 			for instanceID := range trackedInstances {
-				var stopTracking bool
+
+				stopTracking := false
+
+				logData := log.Data{
+					"instanceID": instanceID,
```
Comment (gedge) [inline cmd/dp-import-tracker/main.go]: `"instance_id": instanceID,`  🐍

File: README.md [hunk 2]
```diff
@@ -27,18 +27,26 @@ for the parent import job, if all other instances are also `completed`).
...
+| KAFKA_ADDR                                  | `localhost:9092`                        | A list of kafka brokers
+| IMPORT_API_ADDR                             | `http://localhost:21800`                | The address of Import API
+| IMPORT_API_AUTH_TOKEN                       | "FD0108EA-825D-411C-9B1D-41EF7727F465"  | Authentication token for access to import API
+| DATASET_API_ADDR                            | `http://localhost:22000`                | The address of Dataset API
+| DATASET_API_AUTH_TOKEN                      | "FD0108EA-825D-411C-9B1D-41EF7727F465"  | Authentication token for access to Dataset API
+| BIND_ADDR                                   | `:21300`                                | address to listen on for healthcheck requests
+| GRACEFUL_SHUTDOWN_TIMEOUT                   | `5s`                                    | how much grace time to allow when shutting down
+| DATABASE_ADDRESS                            | `bolt://localhost:7687`                 | The address of the database
+
+HierarchyBuiltTopic         string        `envconfig:"HIERARCHY_BUILT_TOPIC"`
```
[the remark this answers is not in the document]
Comment (CarlHembrough) [reply README.md]: 👍

File: api/dataset_api.go [hunk 3]
```diff
@@ -35,11 +35,30 @@ type InstanceResults struct {
...
-	TotalInsertedObservations int64         `json:"total_inserted_observations,omitempty"`
-	State                     string        `json:"state"`
+	InstanceID           string               `json:"id"`
+	Links                InstanceLinks        `json:"links,omitempty"`
+	NumberOfObservations int64                `json:"total_observations"`
+	State                string               `json:"state"`
+	ImportTasks          *InstanceImportTasks `json:"import_tasks"`
+}
+
+// InstanceImportTasks
```
[the remark this answers is not in the document]
Comment (CarlHembrough) [reply api/dataset_api.go]: 👍

File: api/dataset_api.go [hunk 4]
```diff
@@ -88,7 +107,16 @@ func (api *DatasetAPI) GetInstances(ctx context.Context, vars url.Values) (insta
 	return instanceResults.Items, isFatal, nil
 }
 
-// UpdateInstanceWithNewInserts tells the Dataset API of a number of observationsInserted for instanceID
+func (api *DatasetAPI) SetImportObservationTaskComplete(ctx context.Context, instanceID string) (isFatal bool, err error) {
```
[the remark this answers is not in the document]
Comment (CarlHembrough) [reply api/dataset_api.go]: 👍

File: api/dataset_api.go [hunk 5]
```diff
@@ -97,6 +125,16 @@ func (api *DatasetAPI) UpdateInstanceWithNewInserts(ctx context.Context, instanc
...
 }
 
+func (api *DatasetAPI) UpdateInstanceWithHierarchyBuilt(ctx context.Context, instanceID, dimensionID string) (isFatal bool, err error) {
+	path := api.url + "/instances/" + instanceID + "/import_tasks/build_hierarchies/" + dimensionID
+	logData := log.Data{"url": path}
+	jsonUpload := []byte(`{"state":"completed"}`)
+	logData["jsonUpload"] = jsonUpload
+	jsonBody, httpCode, err := api.put(ctx, path, jsonUpload)
+	logData["jsonBytes"] = jsonBody
+	return errorChecker("SetImportObservationTaskComplete", err, httpCode, &logData)
```
[the remark this answers is not in the document]
Comment (CarlHembrough) [reply api/dataset_api.go]: 👍

File: cmd/dp-import-tracker/main.go (hunk 1 above)
Comment (CarlHembrough) [reply cmd/dp-import-tracker/main.go]: 👍

**Comment by gedge** [comment]
lgtm

**Pull request merged**